public/js/directives/fileModel.js

```diff
@@ -4,34 +4,53 @@ app.directive('fileList', ['$http', '$localStorage', function($http, $localStora
   return {
     restrict: 'EA',
 
-    link: function(scope, element, attrs) {
-      element.bind('change', function(evt) {
-        scope.$apply(function() {
+    link(scope, element, attrs) {
+      element.bind('change', (evt) => {
+        scope.$apply(() => {
           scope[attrs.name] = evt.target.files;
           scope.fileReady = true;
+
           // new fileReader object
-          var reader = new FileReader();
+          const reader = new FileReader();
+
           // inject file on onload
           reader.onload = function(event) {
-            scope.$apply(function() {
-              scope.files.push(evt.target.files[0].name);
-              fileName = evt.target.files[0].name;
-              the_url = event.target.result;
-              createIndex(fileName, the_url);
+            scope.$apply(() => {
+              const fileContent = event.target.result;
+
+              // Checks if file is a valid JSON file
+              if (isJSON(fileContent)) {
+                scope.files.push(evt.target.files[0].name);
+                let fileName = evt.target.files[0].name;
+                $localStorage[fileName] = fileContent;
+                scope.showError = false;
+                scope.noFiles = false;
+              } else {
+                scope.error = 'Not a valid JSON file';
+                scope.showError = true;
+              }
             });
           };
           // when file is read it triggers the onload event
           reader.readAsText(evt.target.files[0]);
         });
       });
-      let createIndex = (fileName, url) => {
-        console.log(url);
-        $http.post('/createIndex', url).success(function(file) {
-          $localStorage[fileName] = file;
-        }).error(function(data) {
-          alert('error');
-        });
+      let isJSON = (fileContent) => {
```

Comment:
### Critical

@andela-wmaina Is there any value to having this method duplicated both on the front and server side?